File: Lab5_13_2/Moving_Material_Point_X.py
import math
from Moving_Material_Point import Moving_Material_Point


class Moving_Material_Point_X(Moving_Material_Point):

    def __init__(self, x0: float, a1: float) -> None:
        self.__x0 = x0
        self.__a1 = a1

    def get_x0(self):
        return self.__x0

    def set_x0(self, value: float):
        self.__x0 = value

    def get_a1(self):
        return self.__a1

    def set_a1(self, value: float):
        self.__a1 = value

    def find_x(self, t: float):
        return self.__x0 + self.__a1 * math.sin(t)

    def find_y(self, t: float):
        return 0

File: Lab5_13_2/Lab5_13_2/Lab5_13_2/MovingMaterialPointX.cs
using System;
namespace Lab5_13_2
{
    public class MovingMaterialPointX: MovingMaterialPoint
    {
        public double X0 { get; set; }
        public double A1 { get; set; }
        public MovingMaterialPointX(double x, double a)
        {
            this.X0 = x;
            this.A1 = a;
        }
        public override double findX(double t)
        {
            return this.X0 + this.A1 * Math.Sin(t);
        }
        public override double findY(double t)
        {
            return 0;
        }
    }
}
